java/src/Ice/src/main/java/com/zeroc/IceSSL/SSLEngine.java

```diff
@@ -1022,7 +679,8 @@ void verifyPeer(String address, ConnectionInfo info, String desc)
             }
         }
 
-        if(_verifyDepthMax > 0 && info.certs != null && info.certs.length > _verifyDepthMax)
+        // Verify depath max includes the root CA, Java doesn't provide it in the certificate chain.
+        if(_verifyDepthMax > 0 && info.certs != null && info.certs.length >= _verifyDepthMax)
```

Comment:
There is a minor discrepancy here with previous behavior, because the custom chain builder added the root CA to the chain returned by Java and we now directly use the chain returned by Java.